src/test/java/ua/chernonog/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package ua.chernonog.onlinebookstore.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.BOOKS_BY_CATEGORY_URL;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORIES_BASE_URL;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_BY_ID_URL;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_DESCRIPTION;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_NAME;
+import static ua.chernonog.onlinebookstore.config.util.UtilConstants.JSON_PATH_FIRST_ITEM_EXISTS;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import ua.chernonog.onlinebookstore.dto.request.category.CreateCategoryDto;
+import ua.chernonog.onlinebookstore.dto.response.book.BookDtoWithoutCategoryIds;
+import ua.chernonog.onlinebookstore.dto.response.category.CategoryResponseDto;
+import ua.chernonog.onlinebookstore.service.CategoryService;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
```

Comment:
For me - better do not mock service layer and make controller tests - integration tests.